posix/clone-dir

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+if [ ! -d .git ]; then
+	set -x
+	git init
+	git remote add origin ${DRONE_REMOTE_URL}
+	set +x
+fi
+
+set -e
+set -x
+
+git config core.sparseCheckout true
```

Comment:
This is not a generic solution. There are a couple of issues present in it.
1. If you want to clone to a separate directory, use DRONE_WORKSPACE env.
2. It assumes that branch to fetch is always master.
3. Handling sparse checkout for pull request or tag is not handled.
4. This does not take care DRONE_COMMIT_REF as well.
5. Changes for windows are not present.

Any change in this repository or (harness managed drone plugins) requires complete support of the feature set.

Reply:
It does not handle plugin depth